### PR TITLE
dvovk/networkspeedtest

### PR DESCRIPTION
**Pull request by dvovk**
Implemented a network speed test to run every 180 seconds. This information is crucial for diagnostic tools. It helps us identify if a client is reaching their maximum capacity.

### CODE REVIEW

File: erigon-lib/diagnostics/client.go [hunk 1]
```diff
@@ -50,6 +52,7 @@ func (d *DiagnosticClient) Setup() {
 	d.setupHeadersDiagnostics(rootCtx)
 	d.setupBodiesDiagnostics(rootCtx)
 	d.setupResourcesUsageDiagnostics(rootCtx)
+	d.setupSpeedtestDiagnostics()
```
Comment (battlmonstr) [inline erigon-lib/diagnostics/client.go:55]:
```suggestion
	d.setupSpeedtestDiagnostics(rootCtx)
```

File: erigon-lib/diagnostics/speedtest.go [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+	ticker := time.NewTicker(180 * time.Second)
+	quit := make(chan struct{})
+	go func() {
+		for {
+			select {
+			case <-ticker.C:
+				d.networkSpeedMutex.Lock()
+				d.networkSpeed = d.runSpeedTest()
+				d.networkSpeedMutex.Unlock()
+			case <-quit:
```
Comment (battlmonstr) [inline erigon-lib/diagnostics/speedtest.go:23]:
```suggestion
			case <-ctx.Done():
```

File: erigon-lib/diagnostics/speedtest.go [hunk 3]
```diff
@@ -0,0 +1,67 @@
+package diagnostics
+
+import (
+	"time"
+
+	"github.com/showwin/speedtest-go/speedtest"
+)
+
+func (d *DiagnosticClient) setupSpeedtestDiagnostics() {
```
Comment (battlmonstr) [inline erigon-lib/diagnostics/speedtest.go:9]:
```suggestion
func (d *DiagnosticClient) setupSpeedtestDiagnostics(ctx context.Context) {
```

**Review by battlmonstr — APPROVED** [review]

File: erigon-lib/diagnostics/client.go (hunk 1 above)
Comment (dvovk) [reply erigon-lib/diagnostics/client.go:55]: fixed

File: erigon-lib/diagnostics/speedtest.go (hunk 2 above)
Comment (dvovk) [reply erigon-lib/diagnostics/speedtest.go]: fixed

File: erigon-lib/diagnostics/speedtest.go (hunk 3 above)
Comment (dvovk) [reply erigon-lib/diagnostics/speedtest.go]: fixed

**Pull request merged**